api/app/Policies/UserPolicy.php

```diff
@@ -68,19 +65,20 @@ public function create(User $user)
      * @param  \App\Models\User  $model
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function update(User $user, User $model, array $injected)
+    public function update(User $user, User $model, ?array $injected = null)
     {
         /**
          * If a user is assigning a role here, check all actions
          * and fail early
          */
-        if (isset($injected["roles"])) {
+        if ($injected && isset($injected["roles"])) {
             if (!$user->isAbleTo("assign-any-role")) {
                 return false;
             }
         }
-
-        return $user->isAdmin() || $user->id === $model->id;
+        // TODO: Right now, for a user to assign-any-role they ALSO need to be able to update-any-user! That doesn't quite match the permissions table.
+        return $user->isAbleTo('update-any-user')
+            || ($user->isAbleTo('update-own-user') && $user->id === $model->id);
```

Comment:
Just confirming, is this a future TODO or were you planning on getting to this in this PR?